src/core/api/clock.ts

```diff
@@ -345,15 +358,25 @@ export interface IClockOptions {
 function createClock(
   mediaElement : HTMLMediaElement,
   options : IClockOptions
-) : Observable<IClockTick> {
-  return observableDefer(() : Observable<IClockTick> => {
+) : IClockHandler {
+  let internalSeek = false;
+  function setCurrentTime(time: number) {
+    internalSeek = true;
+    mediaElement.currentTime = time;
+  }
+  const clock$ = observableDefer(() : Observable<IClockTick> => {
     let lastTimings : IClockTick = objectAssign(
       getMediaInfos(mediaElement, "init"),
-      { stalled: null, getCurrentTime: () => mediaElement.currentTime });
-
+      { stalled: null,
+        getCurrentTime: () => mediaElement.currentTime }
+    );
     function getCurrentClockTick(state : IMediaInfosState) : IClockTick {
       const mediaTimings = getMediaInfos(mediaElement, state);
-      const stalledState = getStalledStatus(lastTimings, mediaTimings, options);
+      const stalledState = getStalledStatus(lastTimings, mediaTimings,
+                                            options, internalSeek);
+      if (state !== "seeking" && internalSeek) {
```

Comment:
Isn't the right condition:

```ts
       if (state === "seeking" && internalSeek) {
```
?

Reply:
Hm not sure to understand clearly, for me an `internalSeek` could only happen in case of an event `seeking` right? so by switching `internalSeek` boolean to false when the event is not `seeking` would do the thing we need right?